### PR TITLE
Add Torch Finance - StableSwap and Weighted Pool on TON blockchain

### PR DESCRIPTION
**Pull request by 0xton**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):  
`Torch Finance`

##### Twitter Link:  
[https://x.com/TorchTon](https://x.com/TorchTon)

##### Website Link:  
[https://torch.finance/](https://torch.finance/)

##### Logo (High resolution, will be shown with rounded borders):  
https://drive.google.com/file/d/1XVXzcVYuvXyIbFtbJLf0uavNd8Jk5GtB/view?usp=sharing

##### Treasury Addresses (if the protocol has treasury): 
[EQCaEOMOR2SRcXTVSolw--rY62ghCoCRjn4Is3bBdnqYwIVZ
](https://tonviewer.com/EQCaEOMOR2SRcXTVSolw--rY62ghCoCRjn4Is3bBdnqYwIVZ)

##### Chain: 
TON

##### Short Description (to be shown on DefiLlama): 
All in one DEX on TON blockchain. Supporting ultra-slow slippage StableSwap and multi-token Weighted Pool all with in one DEX. The Open League Hackathon Top 3.

##### Category (full list at https://defillama.com/categories) *Please choose only one:
`Dexes`

##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):
we do not use oracle

##### Implementation Details: Briefly describe how the oracle is integrated into your project:
we do not use oracle

##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:
we do not use oracle

##### forkedFrom (Does your project originate from another project): 
we are original, not forked from any project.

##### methodology (what is being counted as tvl, how is tvl being calculated):
We have a router contract which accounts for storing all assets, and it will delegate to pools for exchange calculation (such as swap). If the token does not exist in our pool, we will refund it directly. So we can simply sum up all assets in Router contract for TVL calculation.  

##### Github org/user (Optional, if your code is open source, we can track activity): 
[https://github.com/torch-core](https://github.com/torch-core)
We will open source our contract after auditing of tokenomics.

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/torch exports TVL: 
        
 
 
    ton                       32.93 k
    
    total                    32.93 k

File: projects/torch/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const ADDRESSES = require("../helper/coreAssets.json")
+const { sumTokensExport } = require('../helper/chain/ton')
+
+ROUTER = 'EQCaEOMOR2SRcXTVSolw--rY62ghCoCRjn4Is3bBdnqYwIVZ'
```
Comment (g1nt0ki) [inline projects/torch/index.js:4]: * const ROUTER =

File: projects/torch/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+module.exports = {
+  timetravel: false,
+  ton: {
+    tvl: sumTokensExport({
+      owners: [ROUTER],
+      tokens: ADDRESSES.ton.TON,
```
Comment (g1nt0ki) [inline projects/torch/index.js:11]: can remove this line

**Comment by llamatester** [comment]
The adapter at projects/torch exports TVL: 
        
 
 
    ton                       32.83 k
    
    total                    32.83 k

**Comment by g1nt0ki** [comment]
thanks for the PR @0xton

**Comment by 0xton** [comment]
Thanks, you are super efficient!! So basically TORCH FINANCE will list on DeFillama after 24 hours right?

**Pull request merged**